src/main/java/de/tum/cit/ase/ares/api/aop/java/instrumentation/pointcut/JavaInstrumentationPointcutDefinitions.java

```diff
@@ -0,0 +1,110 @@
+                        .toArray(String[]::new)
+        );
+
+    }
+
+    public static ElementMatcher<NamedElement> getClassesMatcher(
+            Map<String, List<String>> methodsMap
+    ) {
+        Set<String> classes = methodsMap.keySet();
+        if (classes.isEmpty()) {
+            return ElementMatchers.none();
+        }
+        return ElementMatchers.namedOneOf(
+                classes
+                        .stream()
+                        .distinct()
+                        .toArray(String[]::new)
+        );
+
+    }
+
+    //</editor-fold>
+
+    //<editor-fold desc="Read Path">
+    public static final Map<String, List<String>> methodsWhichCanReadFiles = Map.of(
+            "instrumentation.io.FileInputStream",
+            List.of("<init>", "read"),
+            "instrumentation.io.RandomAccessFile",
+            List.of("read", "readFully", "readLine", "readBoolean", "readByte", "readChar", "readDouble",
+                    "readFloat", "readInt", "readLong", "readShort", "readUnsignedByte", "readUnsignedShort"),
+            "instrumentation.io.UnixFileSystem",
+            List.of("getLastModifiedTime", "getBooleanAttributes0", "getSpace", "canonicalize0"),
+            "instrumentation.io.WinNTFileSystem",
+            List.of("getBooleanAttributes", "canonicalize", "getLastModifiedTime", "getSpace"),
+            "instrumentation.io.Win32FileSystem",
+            List.of("getBooleanAttributes", "canonicalize", "getLastModifiedTime", "getSpace"),
+            "java.nio.file.Files",
+            List.of("readAttributes", "readAllBytes", "readAllLines", "readString", "read", "newInputStream", "lines")
+    );
+    //</editor-fold>
+
+    //<editor-fold desc="Overwrite Path">
+    public static final Map<String, List<String>> methodsWhichCanOverwriteFiles = Map.of(
+            "java.io.FileOutputStream",
+            List.of("<init>","write"),
+            "instrumentation.io.RandomAccessFile",
+            List.of("write", "writeBoolean", "writeByte", "writeBytes",
+                    "writeChar", "writeChars", "writeDouble", "writeFloat", "writeInt", "writeLong", "writeShort"),
+            "instrumentation.io.UnixFileSystem",
+            List.of("setLastModifiedTime", "createFileExclusively", "delete0", "createDirectory"),
+            "instrumentation.io.WinNTFileSystem",
+            List.of("createFileExclusively", "delete", "setLastModifiedTime", "createDirectory"),
+            "instrumentation.io.Win32FileSystem",
+            List.of("createFileExclusively", "delete", "setLastModifiedTime", "createDirectory"),
+            "instrumentation.util.prefs.FileSystemPreferences",
+            List.of("lockFile0", "unlockFile0")
+    );
+    //</editor-fold>
+
+    //<editor-fold desc="Execute Path">
+    public static final Map<String, List<String>> methodsWhichCanExecuteFiles = Map.of(
+            // TODO why do we have instrumentation I reverted them back to java but still the methods are not intercepted
```

Comment:
same here

Reply:
Addressed in [e5a7f84a664b8e666a8cd00b5e15224a6946da2d](https://github.com/ls1intum/Ares2/pull/25/commits/e5a7f84a664b8e666a8cd00b5e15224a6946da2d)